### DISCUSSION TITLE
How to make nvchad stop messing with indentations when commenting?

### DISCUSSION THREAD

**Discussion by framegrace** [NvChad/NvChad#2618] [Q&A] [answered]
When adding a "#" on a code line (Happens on all languages i try), nvchad adds identations to the line (sometimes removes the front indentation).
Appart of not seeing the utility of that, and why is set up by default, It messes all my yaml file structure and I guess python users will be also mad about this too.

I discovered that `:set indentkeys-=0#` fixes the issue... , so far, so good. 
But how do I make nvchad to do that automatically on start? I tried all I could and modified init.lua and custom/init.lua  with all the different options I've been able to to find but no luck.
Ive tried adding all this different ways  and placed them on init.lua and lua/custom/init.lua with no luck:
- vim.opt.indentkeys:remove("0#")
- vim.cmd('set indentkeys-=0#')
- vim.cmd [[
augroup MyAutoCmds
    autocmd!
    autocmd VimEnter * set indentkeys-=0#
augroup END
]]
- vim.api.nvim_create_autocmd("VimEnter", {
    pattern = "*",
    command = "set indentkeys-=0#",
})

None of them worked, nor any variation or combination of those.

Is that hard to just say to nvim "After doing all your things, just before giving control to the user, do :set indentkeys-=0#
Please help me because this is turning me mad.
Have big yaml files and sometimes need to comment or uncomment pieces of it. If I try that with nvchad it destroys all the structure that can be quite some levels deep.

**Comment by siduck** [comment, marked as answer]
put it in custom init.lua please. Only if you read our docs you wouldve known that general vimscript/lua code can just go in the custom/init.lua. 

also please use vim.opt instead of vim.cmd

**Comment by framegrace** [comment]
Tried the general init.lua just out of frustation. Already used the custom/init.lua to remove mouse interaction, but this particular setting is just not working.
I'm doing anything wrong?
this is my cat  .config/nvim/lua/custom/init.lua

```
vim.opt.mouse=nil
vim.opt.indentkeys:remove("0#")
```

**Reply by siduck** [reply]
i tried your code man, it works here. Please seek help in our discord server as some could help you there

**Comment by woliver99** [comment]
After 2 hours I finally figured out that `vim.opt.smartindent = true` was causing the indentation to disappears when you type `#` on a new line right after it was made.
